internal/command/deploy/deployv2.go

```diff
@@ -0,0 +1,388 @@
+
+	if cliBuildSecrets != nil {
+		opts.BuildSecrets = cliBuildSecrets
+	}
+
+	var buildArgs map[string]string
+	if buildArgs, err = mergeBuildArgs(ctx, build.Args); err != nil {
+		return
+	}
+
+	opts.BuildArgs = buildArgs
+
+	if opts.DockerfilePath, err = resolveDockerfilePathV2(ctx, appConfig); err != nil {
+		return
+	}
+
+	if opts.IgnorefilePath, err = resolveIgnorefilePathV2(ctx, appConfig); err != nil {
+		return
+	}
+
+	if target := appConfig.DockerBuildTarget(); target != "" {
+		opts.Target = target
+	} else if target := flag.GetString(ctx, "build-target"); target != "" {
+		opts.Target = target
+	}
+
+	// finally, build the image
+	heartbeat := resolver.StartHeartbeat(ctx)
+	defer resolver.StopHeartbeat(heartbeat)
+	if img, err = resolver.BuildImage(ctx, io, opts); err == nil && img == nil {
+		err = errors.New("no image specified")
+	}
+
+	if err == nil {
+		tb.Printf("image: %s\n", img.Tag)
+		tb.Printf("image size: %s\n", humanize.Bytes(uint64(img.Size)))
+	}
+
+	return
+}
+
+func resolveDockerfilePathV2(ctx context.Context, appConfig *appv2.Config) (path string, err error) {
+	defer func() {
+		if err == nil && path != "" {
+			path, err = filepath.Abs(path)
+		}
+	}()
+
+	if path = appConfig.Dockerfile(); path != "" {
+		path = filepath.Join(filepath.Dir(appConfig.FlyTomlPath), path)
+	} else {
+		path = flag.GetString(ctx, "dockerfile")
+	}
+
+	return
+}
+
+func resolveIgnorefilePathV2(ctx context.Context, appConfig *appv2.Config) (path string, err error) {
+	defer func() {
+		if err == nil && path != "" {
+			path, err = filepath.Abs(path)
+		}
+	}()
+
+	if path = appConfig.Ignorefile(); path != "" {
+		path = filepath.Join(filepath.Dir(appConfig.FlyTomlPath), path)
+	} else {
+		path = flag.GetString(ctx, "ignorefile")
+	}
+
+	return
+}
+
+func fetchImageRefV2(ctx context.Context, cfg *appv2.Config) (ref string, err error) {
+	if ref = flag.GetString(ctx, "image"); ref != "" {
+		return
+	}
+
+	if cfg != nil && cfg.Build != nil {
+		if ref = cfg.Build.Image; ref != "" {
+			return
+		}
+	}
+
+	return ref, nil
+}
+
+func createReleaseV2(ctx context.Context, appConfig *appv2.Config, img *imgsrc.DeploymentImage) (*api.Release, *api.ReleaseCommand, error) {
+	tb := render.NewTextBlock(ctx, "Creating release")
+
+	input := api.DeployImageInput{
+		AppID: appConfig.AppName,
+		Image: img.Tag,
+	}
+
+	// Set the deployment strategy
+	if val := flag.GetString(ctx, "strategy"); val != "" {
+		input.Strategy = api.StringPointer(strings.ReplaceAll(strings.ToUpper(val), "-", "_"))
+	}
+
+	definition, err := appConfig.ToDefinition()
+
+	if err != nil {
+		return nil, nil, err
+
+	}
+
+	if len(*definition) > 0 {
+		input.Definition = api.DefinitionPtr(*definition)
+	}
+
+	// Start deployment of the determined image
+	client := client.FromContext(ctx).API()
+
+	release, releaseCommand, err := client.DeployImage(ctx, input)
+	if err == nil {
+		tb.Donef("release v%d created\n", release.Version)
+	}
+
+	return release, releaseCommand, err
+}
```

Comment:
It's really pedantic, but CI wants there to be a \n at the end of the file I think

Reply:
Lol i saw